### PR TITLE
Update docker file (for latest nsys)

### PR DESCRIPTION
**Pull request by pgrete**
## PR Summary

The updated cuda version on the host seems to have resulted in the docker `nsys` stalling (with tests timing out after three hours of "finalizing").
This updates also the docker image to use Cuda 11.3 with the latest version of `nsys`.
For reference: We use `nsys profile` to ensure there are not unintended host to device or vice versa copies during execution.

## PR Checklist

- [x] Changes are summarized in CHANGELOG.md

### CODE REVIEW

**Comment by pgrete** [comment]
I now also updated the thresholds for the Overlapping SpaceInstances so that they are less restrictive (but still check the SpaceInstances themselves work), see also #489

**Review by Yurlungur — APPROVED** [review]

**Comment by pgrete** [comment]
Just for reference: I manually triggered the full test suite and it passed.

File: tst/unit/kokkos_abstraction.cpp [hunk 1]
```diff
@@ -622,8 +622,9 @@ struct SmallNLongTBufferPack {
   template <typename TimeType>
   static void test_time(const TimeType time_default, const TimeType time_spaces,
                         const int nspaces) {
```
Comment (jlippuner) [inline tst/unit/kokkos_abstraction.cpp:624]: So we don't need `nspaces` anymore? In that case we should do:
```suggestion
                        const int /*nspaces*/) {
```

**Review by jlippuner — APPROVED** [review]
Looks good. One tiny comment.

File: tst/unit/kokkos_abstraction.cpp (hunk 1 above)
Comment (pgrete) [reply tst/unit/kokkos_abstraction.cpp:624]: Will handle that in a separate PR not not trigger CI again.

Comment (felker) [reply tst/unit/kokkos_abstraction.cpp:624]: Curious, is this because of `-Wunused-parameter` ? Why not remove the `const int` too?

**Pull request merged**